src/vehicle.cpp

```diff
@@ -3413,57 +3421,55 @@ void vehicle::noise_and_smoke( double load, double time )
 
     for( size_t e = 0; e < engines.size(); e++ ) {
         int p = engines[e];
-        if( is_engine_on(e) &&
-                (is_engine_type(e, fuel_type_muscle) || fuel_left (part_info(p).fuel_type)) ) {
+        // FIXME: fuel_left should be called with the vehicle_part's fuel_type for flexfuel support
+        if( is_engine_on( e ) &&
+            ( is_engine_type( e, fuel_type_muscle ) || fuel_left( part_info( p ).fuel_type ) ) ) {
             double pwr = 10.0; // Default noise if nothing else found, shouldn't happen
-            double max_pwr = double(power_to_epower(part_power(p, true)))/40000;
+            double max_pwr = double( power_to_epower( part_power( p, true ) ) ) / 40000;
```

Comment:
Consider splitting off `power_to_epower` into its own variable. Calculation of its parameter `part_power(...)` is complex enough to be worth saving independently. Calculated a second time to assign variable `double j` in combustion branch.

Reply:
Fixed. Gah that code is confusing.